src/Guides/RestructuredText/HTML/Directives/AbstractAdmonitionDirective.php

```diff
@@ -39,18 +39,23 @@ final public function processSub(
         array $options
     ) : ?Node {
         $environment = $document->getEnvironment();
-        $wrapperDiv = function () use ($environment, $options) {
-            return $environment->getRenderer()->render(
-                'directives/admonition.html.twig',
-                [
-                    'name' => $this->name,
-                    'text' => $this->text,
-                    'class' => $options['class'] ?? null,
-                ]
-            );
-        };
 
-        return $parser->getNodeFactory()->createRawNode($wrapperDiv);
+        return $parser
+            ->getNodeFactory()
+            ->createWrapperNode(
+                $document,
+                function () use ($environment, $options) {
+                    return $environment->getRenderer()->render(
+                        'directives/admonition.html.twig',
+                        [
+                            'name' => $this->name,
+                            'text' => $this->text,
+                            'class' => $options['class'] ?? null,
+                        ]
+                    );
+                },
+                '</article></div>'
```

Comment:
I guess the reason why this is here is because directives can be nested?

Reply:
Unfortunately yes .. I would have loved a different approach but the Guides library does nesting with callables and promises; which make things a bit oddish. And I do not have enough knowledge on the internals yet to make a solution that matches my expectations better